timeseries/src/autogluon/timeseries/models/chronos/pipeline/chronos_bolt.py

```diff
@@ -449,16 +449,23 @@ def predict_quantiles(
             .swapaxes(1, 2)
         )
 
-        quantile_levels = [str(q) for q in quantile_levels]
+        quantile_levels_labels = [str(q) for q in quantile_levels]
 
-        if set(quantile_levels).issubset(set(self.quantiles)):
+        if set(quantile_levels_labels).issubset(set(self.quantiles)):
             # no need to perform intra/extrapolation
-            quantiles = predictions[..., [self.quantiles.index(q) for q in quantile_levels]]
-            mean = predictions[:, :, self.quantiles.index("0.5")]
-            return quantiles, mean
+            quantiles = predictions[..., [self.quantiles.index(q) for q in quantile_levels_labels]]
         else:
-            # TODO: Handle interpolation for arbitrary quantiles
-            raise NotImplementedError()
+            # we rely on torch for interpolating quantiles if quantiles that
+            # Chronos Bolt was trained on were not provided
+            augmented_predictions = torch.cat(
+                [predictions[..., [0]], predictions, predictions[..., [-1]]],
```

Comment:
This piece of code seems to linearly interpolate between the P10 and P90 predictions, so the interpolated P51 might be far from the predicted P50. Maybe we can use pandas for interpolation instead?
```python
import pandas as pd

data = {
    0.1: [1, 2, 3],
    0.2: [1.5, 2.5, 3.5],
    0.3: [2, 3, 4],
    0.4: [2.5, 3.5, 4.5],
    0.5: [3, 4, 5],
    0.6: [3.5, 4.5, 5.5],
    0.7: [4, 5, 6],
    0.8: [4.5, 5.5, 6.5],
    0.9: [5, 6, 7]
}
df = pd.DataFrame(data)
query_quantiles = [0.05, 0.15, 0.25, 0.5, 0.75]

df = df.reindex(columns=sorted(set(df.columns).union(query_quantiles))).interpolate(axis=1)
df = df.bfill(axis=1).ffill(axis=1)  # Extrapolate quantiles outside of the [0.1, 0.9] range by "clipping"
result_df = df[query_quantiles]
```